File: problem53/python/problem53.py
def fact(n):
    result = 1
    start = 1
    while (start <= n):
        result *= start
        start += 1
    return result


def combinatorics(x):
    return lambda y : fact(x) / fact(y) / fact(x - y)

print sum(map((lambda y : len(filter(lambda x : x > 1000000,map(combinatorics(y),range(1,y + 1))))),range(1,101)))

File: problem53/haskell/problem53.hs
fact :: (Eq a, Num a) => a -> a
fact 0  = 1
fact n  = n * fact (n - 1)

combinatorics :: Integral a => a -> a -> a
combinatorics x y = fact x `div` fact y `div` fact (x - y)

countNumber :: [Integer] -> Int
countNumber = sum . map (\y -> length . filter (\x -> x > 1000000) . map (combinatorics y) $ [1..y])
